src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -488,14 +488,24 @@ public void cancelOrders(List<Order> targets) {
     //@@ author kangtinglee
     @Override
     public boolean canFulfilOrder(Order target) {
-        List<Pair<Dish, Integer>> dishesList = target.getDishQuantityList();
+        List<Order> temp = new ArrayList<>();
+        temp.add(target);
+        return canFulfilOrders(temp);
+    }
+
+    //@@ author kangtinglee
+    @Override
+    public boolean canFulfilOrders(List<Order> orders) {
         HashMap<Ingredient, Integer> ingredientsTable = new HashMap<>();
-        for (Pair<Dish, Integer> p : dishesList) {
-            for (Pair<Ingredient, Integer> d : p.getKey().getIngredientQuantityList()) {
-                if (!ingredientsTable.containsKey(d.getKey())) {
-                    ingredientsTable.put(d.getKey(), 0);
+        for (Order target : orders) {
+            List<Pair<Dish, Integer>> dishesList = target.getDishQuantityList();
+            for (Pair<Dish, Integer> p : dishesList) {
+                for (Pair<Ingredient, Integer> d : p.getKey().getIngredientQuantityList()) {
+                    if (!ingredientsTable.containsKey(d.getKey())) {
+                        ingredientsTable.put(d.getKey(), 0);
+                    }
+                    ingredientsTable.put(d.getKey(), ingredientsTable.get(d.getKey()) + d.getValue() * p.getValue());
```

Comment:
It might be good to reduce the deep nested loop here for code quality purposes state [here](https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)